[Diagnostics.Trace] Extension method for configuring outgoing request trace handler.

amanda-tarafa opened the pull request:
Continuation to #2661.

jskeet commented on apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Trace/TraceTest.cs:
```diff
@@ -383,7 +383,7 @@ public void ConfigureServices(IServiceCollection services)
                 c.BaseAddress = new Uri("https://google.com/");
             })
-            .AddHttpMessageHandler<UnchainedTraceHeaderPropagatingHandler>();
+            .AddGoogleOutgoingTraceHandler();
```
Nit: let's indent this line and the three above.

amanda-tarafa replied:
Done

amanda-tarafa replied:
Done

jskeet commented on apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Snippets/TraceSnippets.cs:
```diff
@@ -216,9 +216,9 @@ public void ConfigureServices(IServiceCollection services)
-            .AddHttpMessageHandler<UnchainedTraceHeaderPropagatingHandler>();
+            // The next call guarantees that outgoing requests made with the HTTP Client being configured
+            // will be traced.
+            .AddGoogleOutgoingTraceHandler();
```
Nit: indent this and comment

amanda-tarafa replied:
Done

amanda-tarafa replied:
Done

jskeet commented on apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Snippets/TraceSnippets.cs:
```diff
@@ -216,9 +216,9 @@ public void ConfigureServices(IServiceCollection services)
-            // that the outgoing requests will be traced.
-            .AddHttpMessageHandler<UnchainedTraceHeaderPropagatingHandler>();
+            // The next call guarantees that outgoing requests made with the HTTP Client being configured
+            // will be traced.
```
I thought it would only add tracing if trace was enabled for the request being handled? Maybe I'm being confused with something else.

amanda-tarafa replied:
No, you are right, changing now.

jskeet commented on apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/HttpClientBuilderExtensions.cs:
```diff
@@ -0,0 +1,35 @@
+    {
+        /// <summary>
+        /// Adds an <see cref="UnchainedTraceHeaderPropagatingHandler"/> to the HTTP client being built,
+        /// which will allow outgoing requests made with this client to be traced.
```
We probably want to word this really carefully.

amanda-tarafa replied:
Reworded.

chrisdunelm commented:
Looks good :) I agree with Jon that `AddOutgoingGoogleTraceHandler` reads slightly better.

amanda-tarafa commented:
Addressed review comments, and yes, the new name is better!

@jskeet PTAL at the rewording of the comment and docstring, I haven't rebased the last commit so it's easier. I'll rebase after and merge if your OK with the changed.

jskeet commented on apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/HttpClientBuilderExtensions.cs:
```diff
@@ -0,0 +1,35 @@
+    {
+        /// <summary>
+        /// Adds an <see cref="UnchainedTraceHeaderPropagatingHandler"/> to the HTTP client being built,
+        /// which will propagate trace information for outgoing request, made with this client, already being traced.
```
"outgoing request" => "outgoing requests" and I'd change "already being traced" to "when the request being handled by this server is already being traced".
